packages/experiment-tag/src/util/cookie.ts

```diff
@@ -0,0 +1,20 @@
+import { CampaignParser } from '@amplitude/analytics-core';
+import { CookieStorage } from '@amplitude/analytics-core';
+import { MKTG } from '@amplitude/analytics-core';
+import type { Campaign } from '@amplitude/analytics-core';
+
+/**
+ * Utility function to generate and set marketing cookie
+ * Parses current campaign data from URL and referrer, then stores it in the marketing cookie
+ */
+export async function setMarketingCookie(apiKey: string) {
+  const storage = new CookieStorage<Campaign>({
+    expirationDays: 365,
+    sameSite: 'Lax',
+  });
+
+  const parser = new CampaignParser();
+  const storageKey = `AMP_${apiKey.substring(0, 10)}_ORIGINAL_${MKTG}`;
```

Comment:
To follow the naming convention:
```suggestion
  const storageKey = `AMP_${MKTG}_ORIGINAL_${apiKey.substring(0, 10)}`;
```
Existing marketing cookie is named as AMP_MKTG_{first ten digits of api key}